src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F

```diff
@@ -528,6 +558,10 @@ subroutine sfclayer_from_MPAS(configs,mesh,diag_physics,sfc_input,its,ite)
              fh_sea(i,j) = fh(i)
              fm_sea(i,j) = fm(i)
           endif
+
+          waterdepth_p(i,j) = 0._RKIND
+          lakedepth_p(i,j)  = 0._RKIND
+          lakemask_p(i,j)   = 0._RKIND
```

Comment:
@mgduda @ldfowler58 This section should be put under 'case ("sf_monin_obukhov_rev") since they are only allocated for that case.

Reply:
Let's me check. Thanks.

Reply:
I did make these corrections a few days ago, following Weis' suggestion. Are you saying that the PR was not updated?
Thanks.

Reply:
Actually I did miss one section. Working on it now.

Reply:
This should solve the issue pointed out by Michael. Running MPAS with the -g option does not reveal any issue when running the original (sf_monin_obukhov) and revised (sf_monin_obukhov_rev) surface layer schemes.